### DISCUSSION TITLE
Subtle Display Flicker or Brightness Flicker

### DISCUSSION THREAD

**Discussion by MeditatedBison** [end-4/dots-hyprland#413] [Q&A] [answered]
when the mouse is still, There is a particular brightness . 
When i move the mouse, Brightness very very subtly decreases, or i don't know what changes, but there is a very subtle more white in the display. 
Like you can know that some change is there

So it just happens with the move movement of mouse

How can i change this thing? Or if its a bug

**Comment by end-4** [comment]
it sounds like your screen's problem
can you send a video?

**Reply by MeditatedBison** [reply]
sadly the screen recorder is not recording that thing, 

What happens, is the brightness reduces by very very small amount, Like maybe 0.5percent when you are moving the mouse!. 

when the mouse is still, there is say `x` brightness, when you move the mouse, the brightness by very very small amount, and again comes back to `x`.

This thing was not there before. i think something in the config of these dotfiles is causing this, Because i today changed back to my old default hyprland config (by changing the config file) this thing was not happening then. But it happens only with these config files

**Reply by MeditatedBison** [reply]
I installed these dotfiles on clean installed arch. And some basic packages 

The problem is still there, and is only with these dotfiles, when i change dotfiles to default hyprland ones, the problem is not there. Are there are any brightness dim config lines in the config files? maybe it's because of that

**Reply by end-4** [reply]
the config doesn't touch your brightness setttings
it just has keybinds for changing brightness

**Comment by MeditatedBison** [comment, marked as answer]
SOLVED!! Solution:
Just turned off `Damage Tracking` option from the Right panel. And that minute yet irritating brightness flicker Vanished!!!!!
![image](https://github.com/end-4/dots-hyprland/assets/95769851/e05be766-1e24-49b5-a692-bafdfe22e459)

**Reply by end-4** [reply]
Ah, I thought that's only a problem for screen recordings
While this may work, it'll result in huge power consumption
What damage tracking does is only track which parts of the screen are updated and only redraw those areas. With this option off, Hyprland will always draw the whole screen on every single frame

**Reply by MeditatedBison** [reply]
Oh, can you give another solution
Like I will give the whole problem in detail, the brightness is changed around the current brightness by +-5 when mouse is moved (I am not sure if it's + or -) but for sure in this range

This is subtle yet attention catching, like with the default Config this issue is not there. So somethung is causing this

My system is a laptop with I5 12500h and rtx 3050 mobile.

**Reply by gouravraghuwanshi** [reply]
did you found any solution

**Comment by MeditatedBison** [comment]
@end-4 The "Damage Tracking" option is turned on whenever i reboot to my system. 
And after using some of my apps, The flicker comes back (brightness dims slightly when you are moving the mouse, or there is some motion on the screen 
But i just turn on And then turn off "Damage Tracking" feature to solve that fliker. 

But can you tell a permanent fix?

**Reply by MeditatedBison** [reply]
I think the issue is due to some shaders or something? If you get an idea about what is causing this, Please tell?
Thank you

**Reply by end-4** [reply]
huh why did i not see this
if you want the damage tracking option to always be off, configure it properly in hyprland config (`debug:damage_tracking`). see the wiki page: https://wiki.hyprland.org/Configuring/Variables/#debug

**Reply by gouravraghuwanshi** [reply]
i have the solution,
This brightness flickering is because of vrr and vfr just change their value 2.
@end-4 @MeditatedBison

**Reply by end-4** [reply]
for vfr, it's a boolean so 2 will be same as 1, which is what i have
for vrr, 2 makes it only enabled when fullscreen. so does the flickering happen when you go fullscreen?

**Reply by gouravraghuwanshi** [reply]
I am sorry for late reply, but yes the flicker is happening in full screen.

**Comment by gouravraghuwanshi** [comment]
Idk I have read it on the wiki of hyprland, it did work for me.

Maybe turning this feature off may help

**Comment by 6iovani** [comment]
I found another fix. I went to the general.conf file in the hyprland folder, and set vrr=0. 
<img width="498" height="325" alt="image" src="https://github.com/user-attachments/assets/97eafc87-eb58-4187-8c03-4db9a1a5c0be" />

**Reply by florianguigue** [reply]
This one does works for me. Thanks!

**Reply by dannightmare** [reply]
This solved it for me too, thanks!
